modules/test/src/test/kotlin/datamaintain/test/AbstractMongoDbTest.kt

```diff
@@ -1,31 +1,63 @@
 package datamaintain.test
 
 import com.mongodb.ConnectionString
+import com.mongodb.client.MongoClient
 import com.mongodb.client.MongoClients
+import com.mongodb.client.MongoCollection
+import com.mongodb.client.MongoDatabase
 import datamaintain.db.driver.mongo.MongoDriver
 import org.bson.Document
 import org.junit.jupiter.api.AfterEach
 import org.junit.jupiter.api.BeforeEach
+import org.testcontainers.containers.MongoDBContainer
+import org.testcontainers.junit.jupiter.Container
+import org.testcontainers.junit.jupiter.Testcontainers
 
+@Testcontainers
 abstract class AbstractMongoDbTest {
+    val mongoTag = "mongo:5.0"
     val databaseName = "datamaintain-test"
-    val mongoUri = "mongodb://localhost:27018/datamaintain-test"
-    val connectionString = ConnectionString(mongoUri)
 
-    private val client = MongoClients.create(connectionString)
-    val database = client.getDatabase(databaseName)
+    @Container
+    val mongoDBContainer: MongoDBContainer = MongoDBContainer(mongoTag)
 
-    val collection = database.getCollection(
-            MongoDriver.EXECUTED_SCRIPTS_COLLECTION, Document::class.java)
+    // Fill when mongo container is up
+    private var mongoUri: String? = null
+    private var connectionString: ConnectionString? = null
+    private var client: MongoClient? = null
+    private var database: MongoDatabase? = null
+    private var collection: MongoCollection<Document>? = null
+
+    fun mongoUri(): String {
+        return mongoUri!!
+    }
+
+    fun database(): MongoDatabase {
+        return database!!
+    }
+
+    fun collection(): MongoCollection<Document> {
+        return collection!!
+    }
 
     @BeforeEach
     fun init() {
-        cleanDb()
+        // Fetch URI of container and start a client on it
+        mongoUri = mongoDBContainer.getReplicaSetUrl(databaseName)
+        connectionString = ConnectionString(mongoUri!!)
+        client = MongoClients.create(connectionString!!)
+        database = client!!.getDatabase(databaseName)
+        collection = database!!.getCollection(MongoDriver.EXECUTED_SCRIPTS_COLLECTION, Document::class.java)
```

Comment:
I'm not fond of putting a `String` value in a `String?` and then ensure the `String?` is not null using !!.